Add PDM to the list of alternatives to Pipenv (#1036)
pypa/packaging.python.org: source/tutorials/managing-dependencies.rst

web applications), but is also very well suited to managing development and
testing environments for any kind of project.

> Alternative dependency management solutions can be found in a non-exhaustive list
> at the bottom of the page.

Reviewer: Let's take advantage of [implicit link targets](https://docs.readthedocs.io/en/stable/guides/cross-referencing-with-sphinx.html#implicit-targets).

```suggestion
For alternatives, see `Other Tools for Application Dependency Management`_.
```

If you find this particular approach to managing application dependencies isn't
working well for you or your use case, you may want to explore these other tools
and techniques, listed in alphabetical order, to see if one of them is a better fit:

* `hatch <https://github.com/ofek/hatch>`_ for opinionated coverage of even

>   more steps in the project management workflow (such as incrementing versions,
>   tagging releases, and creating new skeleton projects from project templates)

Reviewer: ```suggestion
  more steps in the project management workflow, such as incrementing versions,
  tagging releases, and creating new skeleton projects from project templates.
```

> * `micropipenv <https://github.com/thoth-station/micropipenv>`_ is a lightweight
>   wrapper for pip to support requirements.txt, Pipenv and Poetry lock files or
>   converting them to pip-tools compatible output. Designed for containerized
>   Python applications but not limited to them.

Reviewer: ```suggestion
* `micropipenv <https://github.com/thoth-station/micropipenv>`_ for a lightweight
  wrapper around pip that supports ``requirements.txt``, Pipenv and Poetry lock files,
  or converting them to pip-tools compatible output. Designed for containerized
  Python applications, but not limited to them.
```

Author: Applied manually (GitHub wouldn't let me).

> * `pdm <https://github.com/pdm-project/pdm>`_ for a modern Python package management
>   tool supporting `PEP 582 <https://www.python.org/dev/peps/pep-0582/>`_
>   (no virtualenvs: installation of packages in a ``__pypackages__`` folder,
>   *à la* ``node_modules``) and relying on standards such as
>   `PEP 517 <https://www.python.org/dev/peps/pep-0517/>`_ and
>   `PEP 621 <https://www.python.org/dev/peps/pep-0621/>`_.

Reviewer: We prefer to use the `:pep:` role for readability. I also think the parenthetical could be more succinct, and rely on the PEP for details.

```suggestion
* `PDM <https://github.com/pdm-project/pdm>`_ for a modern Python package management
  tool supporting :pep:`582` (replacing virtual environments with ``__pypackages__``
  directory for package installation) and relying on standards such as :pep:`517` and
  :pep:`621`.
```

> * `pip-tools <https://github.com/jazzband/pip-tools>`_ to build your own
>   custom workflow from lower level pieces like ``pip-compile`` and ``pip-sync``

Reviewer: ```suggestion
* `pip-tools <https://github.com/jazzband/pip-tools>`_ for creating a lock file of all
  dependencies from a list of packages directly used in a project, and ensuring that
  only those dependencies are installed.
```

> * `poetry <https://github.com/python-poetry/poetry>`__ for a tool comparable in scope
>   to ``pipenv`` that focuses more directly on use cases where the repository being
>   managed is structured as a Python project with a valid ``pyproject.toml`` file
>   (by contrast, ``pipenv`` explicitly avoids making the assumption that the
>   application being worked on that's depending on components from PyPI will
>   itself support distribution as a ``pip``-installable Python package).

Reviewer: ```suggestion
* `Poetry <https://github.com/python-poetry/poetry>`__ for a tool comparable in scope
  to Pipenv that focuses more directly on use cases where the project being managed is
  structured as a distributable Python package with a valid ``pyproject.toml`` file.
  By contrast, Pipenv explicitly avoids making the assumption that the application
  being worked on will support distribution as a ``pip``-installable Python package.
```

Author: Applied manually (GitHub wouldn't let me).
